From 187ecfb9752ac00736fc5fde0677efbb46c124d3 Mon Sep 17 00:00:00 2001
From: Xilin Wu <sophon@radxa.com>
Date: Tue, 19 May 2026 22:55:30 +0800
Subject: [PATCH 259/268] dt-bindings: net: toshiba,tc956x-dwmac: Allow MAC
 address properties

TC956x Ethernet functions may use a board-provided MAC address, including
one stored in an NVMEM cell.

Allow the standard MAC address properties on the PCI function nodes so
device trees can describe mac-address, local-mac-address, or an NVMEM cell
named "mac-address" without triggering schema warnings.

The binding cannot reference ethernet-controller.yaml wholesale because
the node is a PCI device node, so reference only the relevant common
property definitions.

Signed-off-by: Xilin Wu <sophon@radxa.com>
---
 .../bindings/net/toshiba,tc956x-dwmac.yaml           | 12 ++++++++++++
 1 file changed, 12 insertions(+)

--- a/Documentation/devicetree/bindings/net/toshiba,tc956x-dwmac.yaml
+++ b/Documentation/devicetree/bindings/net/toshiba,tc956x-dwmac.yaml
@@ -39,6 +39,18 @@ properties:
   # contradictory $nodename rules (`ethernet@` versus `pci@`). Happily only a
   # small number of the properties are useful on TC956x so we can just reference
   # what we need.
+  local-mac-address:
+    $ref: /schemas/net/network-class.yaml#/properties/local-mac-address
+
+  mac-address:
+    $ref: /schemas/net/network-class.yaml#/properties/mac-address
+
+  nvmem-cells:
+    $ref: ethernet-controller.yaml#/properties/nvmem-cells
+
+  nvmem-cell-names:
+    $ref: ethernet-controller.yaml#/properties/nvmem-cell-names
+
   phy-connection-type:
     $ref: ethernet-controller.yaml#/properties/phy-connection-type
 
